test/jdk/sun/net/www/protocol/https/ChunkedOutputStream.java

```diff
@@ -57,100 +79,99 @@
+                String chunk = req.getRequestHeaders().getFirst("Transfer-encoding");
                 if (!"chunked".equals(chunk)) {
-                    req.sendResponse(501, "Internal server error");
-                    req.orderlyClose();
+                    req.sendResponseHeaders(501, -1);
+                    break;
                 }
-                req.setResponseEntityBody(reqbody);
                 if (count == 1) {
-                    req.setResponseHeader("Connection", "close");
+                    req.getResponseHeaders().set("Connection", "close");
                 }
-                req.sendResponse(200, "OK");
-                if (count == 1) {
-                    req.orderlyClose();
+                req.sendResponseHeaders(200, 0);
+                try (PrintWriter pw = new PrintWriter(req.getResponseBody())) {
+                    pw.print(reqbody);
                 }
                 break;
             case 2: /* test 3 */
-                reqbody = req.getRequestEntityBody();
+                reqbody = new String(req.getRequestBody().readAllBytes());
                 if (!reqbody.equals(str2)) {
-                    req.sendResponse(500, "Internal server error");
-                    req.orderlyClose();
+                    req.sendResponseHeaders(500, -1);
+                    break;
                 }
-                int clen = Integer.parseInt (
-                        req.getRequestHeader("Content-length"));
+                int clen = Integer.parseInt (req.getRequestHeaders().getFirst("Content-length"));
                 if (clen != str2.length()) {
-                    req.sendResponse(501, "Internal server error");
-                    req.orderlyClose();
+                    req.sendResponseHeaders(501, -1);
+                    break;
+                }
+                req.getResponseHeaders().set("Connection", "close");
+                req.sendResponseHeaders(200, 0);
+                try (PrintWriter pw = new PrintWriter(req.getResponseBody())) {
+                    pw.print(reqbody);
                 }
-                req.setResponseEntityBody (reqbody);
-                req.setResponseHeader("Connection", "close");
-                req.sendResponse(200, "OK");
-                req.orderlyClose();
                 break;
             case 3: /* test 6 */
-                req.setResponseHeader("Location", "https://foo.bar/");
-                req.setResponseHeader("Connection", "close");
-                req.sendResponse(307, "Temporary Redirect");
-                req.orderlyClose();
+                req.getResponseHeaders().set("Location", "https://foo.bar/");
+                req.getResponseHeaders().set("Connection", "close");
+                req.sendResponseHeaders(307, -1);
                 break;
             case 4: /* test 7 */
             case 5: /* test 8 */
-                reqbody = req.getRequestEntityBody();
+                reqbody = new String(req.getRequestBody().readAllBytes());
```

Comment:
Please double check that client and server use the same charset, or better, explicitly use UTF-8 everywhere.